rmi.go

```diff
@@ -0,0 +1,154 @@
+/*
+   Copyright (C) nerdctl authors.
+   Copyright (C) containerd authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+)
+
+var rmiCommand = &cli.Command{
+	Name:      "rmi",
+	Usage:     "Remove one or more images",
+	ArgsUsage: "[flags] IMAGE [IMAGE, ...]",
+	Action:    rmiAction,
+}
+
+func rmiAction(clicontext *cli.Context) error {
+	if clicontext.NArg() == 0 {
+		return errors.Errorf("requires at least 1 argument")
+	}
+
+	client, ctx, cancel, err := newClient(clicontext)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+
+	var (
+		imageStore = client.ImageService()
+		cs         = client.ContentStore()
+	)
+
+	imageList, err := imageStore.List(ctx, "")
+	if err != nil {
+		return err
+	}
+
+	var opts []images.DeleteOpt
+	var imageNotFoundError bool
+
+	if clicontext.NArg() >= 1 {
+		img := clicontext.Args().First()
+		imgFQIN := getFQIN(img)
+
+		digests, err := getImageDigests(ctx, cs, imgFQIN, imageList)
+		if err != nil {
+			return errors.Errorf("Error in getting image digests: %v", err)
+		}
+
+		if err := imageStore.Delete(ctx, imgFQIN, opts...); err != nil {
+			fmt.Printf("Error: No such image: %s\n", img)
+			imageNotFoundError = true
+		} else {
+			printDigests(imgFQIN, digests)
+		}
+	}
+
+	if clicontext.NArg() == 1 {
+		if imageNotFoundError {
+			os.Exit(1)
+		}
+		os.Exit(0)
+	}
+
+	for _, img := range clicontext.Args().Tail() {
+		imgFQIN := getFQIN(img)
+
+		digests, err := getImageDigests(ctx, cs, imgFQIN, imageList)
+		if err != nil {
+			return errors.Errorf("Error in getting image digests: %v", err)
+		}
+
+		if err := imageStore.Delete(ctx, imgFQIN, opts...); err != nil {
+			if errdefs.IsNotFound(err) {
+				fmt.Printf("Error: No such image: %s\n", img)
```

Comment:
logrus.WithError(err).Errorf(...)

Reply:
So I made this change, however when the images are missing the error looks like this:

```
root@vagrant:~/go/src/github.com/nerdctl# nerdctl rmi redis traefik:v2.3.2
ERRO[0000] Error: No such image: redis                   error="image \"docker.io/library/redis:latest\": not found"
ERRO[0000] Error: No such image: traefik:v2.3.2          error="image \"docker.io/library/traefik:v2.3.2\": not found"
```

My reason for using `fmt.Printf()` was to keep the output the same as `docker rmi`. This is how it looks like in `docker rmi`

```
 SM-smahajan628:~ smahajan$ docker rmi traefik redis
Error: No such image: traefik
Error: No such image: redis
SM-smahajan628:~ smahajan$ echo $?
1
```

Reply:
SGTM, but `fmt.Fprintf(clicontext.App.Writer, ...)` would be preferable than plain `fmt.Printf`

Reply:
Fixed.